index.bs

```diff
@@ -968,6 +970,69 @@ spec: ecma-262; urlPrefix: http://tc39.github.io/ecma262/
     portals-close-window.html
   </wpt>
 
+  Navigation {#patch-navigation}
+  ------------------------------
+
+  Patch the <a spec=HTML>navigate</a> algorithm to prevent certain navigations in a
+  portal as follows:
+
+  <div algorithm="navigate patch">
+    In <a spec=HTML>navigate</a>, in the case where <var ignore>resource</var> is a [=request=]
+    whose [=request/url=]'s [=url/scheme=] is "`javascript`", prepend the following step as the
+    first step of the queued task:
+
+    1. If |browsingContext|'s [=portal state=] is not "`none`", then [=close a portal
+        element|close=] |browsingContext|'s [=host element=] and abort these steps.
```

Comment:
Great catch. I forgot `javascript:` in my case analysis, at least mentally, so I didn't properly harmonize it with our desired outcome for the others.

Reply:
So for string-returning JavaScript snippets, the result gets the default 200 status code. I think we want that to close the portal. It's the non-string returning ones that we want to allow.

Reply:
Hmm. Well, what falls out naturally from the existing spec text is a bit different.

If we delete this section entirely, we get the following:

- If the JS returns a string, then it will create a 200 response with null url.
  - If we're in the initial-load case, this will close the portal
  - If we're in the non-initial-load case, then this will bail out and do nothing
- If the JS returns a non-string, then it will create a 204 response with null url.
  - Exact same as in the string case.

I guess that's OK? Maybe this just reflects the bad existing JS spec; e.g. maybe some of those responses should be non-null URL. I can look into that, but perhaps we should just move ahead with this PR by deleting this section.

Reply:
Final (?) reply-to-myself: it looks like the existing spec is correct in this; both string and non-string cases have null response URL.

- http://software.hixie.ch/utilities/js/live-dom-viewer/?saved=8481
- http://software.hixie.ch/utilities/js/live-dom-viewer/?saved=8482

Reply:
I'm not sure whether the response URL is null or not for the generated response (I don't see anywhere that sets it), but `document.URL` is not -- it remains the same as the page it replaced (or possibly the initiator?):

http://software.hixie.ch/utilities/js/live-dom-viewer/?%3C!DOCTYPE%20html%3E%0A%3Ciframe%20src%3D%22data%3Atext%2Fhtml%2C%3Ca%20href%3D%26quot%3Bjavascript%3A%27%3Cscript%3Eonload%20%3D%20()%20%3D%3E%20document.body.appendChild(document.createTextNode(document.URL))%3C%2Fscript%3E%27%26quot%3B%3Eclick%20me%3C%2Fa%3E%22%3E%3C%2Fiframe%3E

Your examples both print document.URL before the synthetic response is generated (because the expression is still being evaluated), and I think they're about:blank because it's the initial about:blank that's executing the javascript: URL in the case where you directly set it as iframe src.

I definitely care way more about non-string ones than string ones (does anyone actually use the string case?), but if it does get to keep the URL of the page it replaces then I would have expected to preserve that, odd as it is.

Reply:
Thanks for correcting my bad examples :). Do you think there's anything we should modify in this PR though? Or is this good to merge?